Adding new weather provider for weather.gov

vincep5 opened the pull request:
I've added a new weather provider for the "new" weather module.  Its an open API from weather.gov a US only provider of weather information.  It might be useful for those that do not want to signup for an account for the other weather providers.

fewieden commented on modules/default/weather/providers/weathergov.js:
```diff
@@ -0,0 +1,270 @@
+					return;
+				}
+
+				//this.setFetchedLocation(`${data.name}, ${data.sys.country}`);
```
Please remove commented code.

fewieden commented on modules/default/weather/providers/weathergov.js:
```diff
@@ -0,0 +1,270 @@
+					return;
+				}
+
+				//this.setFetchedLocation(`${data.city.name}, ${data.city.country}`);
```
Please remove commented code.

fewieden commented on modules/default/weather/providers/weathergov.js:
```diff
@@ -0,0 +1,270 @@
+	generateWeatherObjectFromCurrentWeather(currentWeatherData) {
+		const currentWeather = new WeatherObject(this.config.units);
+
+		//currentWeather.humidity = currentWeatherData.main.humidity;
```
Please remove commented code.

fewieden commented on modules/default/weather/providers/weathergov.js:
```diff
@@ -0,0 +1,270 @@
+		currentWeather.windSpeed = currentWeatherData.windSpeed.split(" ", 1);
+		currentWeather.windDirection = this.convertDirectiontoDegrees(currentWeatherData.windDirection);
+		currentWeather.weatherType = this.convertWeatherType(currentWeatherData.shortForecast, currentWeatherData.isDaytime);
+		//currentWeather.sunrise = moment(currentWeatherData.sys.sunrise, "X");
```
Please remove commented code.

fewieden commented on modules/default/weather/providers/weathergov.js:
```diff
@@ -0,0 +1,270 @@
+		return null;
+	},
+
+	/* getParams(compliments)
```
getParams has no parameter called `compliments`, please remove

fewieden commented on modules/default/weather/providers/weathergov.js:
```diff
@@ -0,0 +1,270 @@
+		// There are way too many types to create, so lets just look for certain strings
+
+		if (weatherType.includes("Cloudy") || weatherType.includes("Partly")) {
+			if (isDaytime) {
```
In this function you can save some else statements:

```js
if (isDaytime) {
  return "day-cloudy";
}

return "night-cloudy";
```

fewieden commented on modules/default/weather/providers/weathergov.js:
```diff
@@ -0,0 +1,270 @@
+		// initial variable declaration
+		const days = [];
+		// variables for temperature range and rain
+		var minTemp = [];
```
please do not mix `var` with `const` and `let` in this function. Stick to `const` and `let` like in the rest of the file.

vincep5 commented:
@fewieden thanks for the review. I have sent up another commit to fix those points you mentioned.  I also fixed openweathermap.js so that const, let, and var declarations are not mixed together.  I removed the var from both files.

fewieden commented:
@vincep5 looks cleaner now
